src/lib/common/include/sol-update-modules.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <stdint.h>
+
+#include "sol-update.h"
+
+#include "sol-macros.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct sol_update {
+#ifndef SOL_NO_API_VERSION
+#define SOL_UPDATE_API_VERSION (1)
+    uint16_t api_version; /**< API version */
+#endif
+
+    struct sol_update_handle * (*check)(const char *url,
+        void (*cb)(void *data, int status,
+        const struct sol_update_info *response),
+        const void *data);
+
+    struct sol_update_handle * (*fetch)(const struct sol_update_info *response,
+        void (*cb)(void *data, int status, const char *file_path),
```

Comment:
You didn't get it. You don't specify the URL or location anywhere in the API. These are "out of band".

In your current implementation these would come from compile-time defines (ie: Kconfig) and for ease of test you can also consume envvars, so you can `export SOL_UPDATE_URL=http://localhost:8080 SOL_UPDATE_FILE=/tmp/bla`. 

Most of the systems will handle this elsewhere, such as in ArchLinux it would be /etc/pacman.conf and mirrors list, Debian is /etc/apt/apt.conf and repositories list, etc.

That's why I said the extension of the `struct sol_update` is an option. But we don't even need to go that route. I guess I wasn't clear about that in my comment :-) (just re-read and got that impression)